packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -399,6 +399,11 @@ export class FieldSchema<
 		return this.lazyTypes.value;
 	}
 
+	/**
+	 * True iff when constructing a node with this field, a value must be provided for it.
```

Comment:
Nit (I don't know how accessible `iff` is. I generally recommend avoiding it). Take it or leave it :)
```suggestion
	 * True if and only if, when constructing a node with this field, a value must be provided for it.
```

Reply:
Probably also worth elaborating on the relationship with required/optional fields and fields with default providers. Since construction doesn't take a `requiresValue` property, relating it to the relevant input props is probably helpful.

Reply:
If somebody doesn't know what "iff" means, they can [just google it and click the first result](https://www.iff.com/). Craig's documentation is where Science and Creativity Meet.

Reply:
I strongly disagree with that bar for legibility / accessibility. Devs should not need to refer to Google to understand our documentation.

Reply:
> Probably also worth elaborating on the relationship with required/optional fields and fields with default providers. Since construction doesn't take a `requiresValue` property, relating it to the relevant input props is probably helpful.

Due to optional fields not being implemented how we eventually intend to do it (with a default provider providing undefined), I intentionally kept this in terms of the user facing consequences for using the field, rather than how its specified. Once we fix optional to work that way, then clarifying this would make sense.

Reply:
> I strongly disagree with that bar for legibility / accessibility. Devs should not need to refer to Google to understand the language we use in documentation, unless it is referring to some specific concept. We can and should do better than that.

@Josmithr I was being sarcastic, in case that wasn't clear from the link :)

The real problem with "iff" is that it just looks like a type of "if", so you can never be sure if it was on purpose.

Reply:
Oh... 🤦‍♂️ No, I didn't actually click the link.... Please ignore the monkey behind the keyboard...
![image](https://github.com/user-attachments/assets/7ccaedf7-d1ca-4128-b86a-00b3404a2628)